File: c_interop/c_lib_no_buildsystem/c_lib.c
// compile the library with `zig cc -c c_lib.c -O3` to produce c_lib.obj

#include <stdint.h>
#include <stdio.h>

#define MY_NUM 4

int32_t square(int32_t x) {
    return x * x;
}

void talk_from_c(){
    printf("Hi! This is being printed from a C library!!\n");
    printf("C says: square of %d is %d\n", MY_NUM, square(MY_NUM));
}

File: c_interop/c_lib_no_buildsystem/c_lib.zig
// run with `zig run c_lib.zig c_lib.obj -lc`
// see c_lib.c to compile the C library

extern fn square(x: i32) i32;
extern fn talk_from_c() void;

const print = @import("std").debug.print;

pub fn main() void {
    const num: i32 = 4;
    print("This is being printed from Zig!!\n", .{});
    print("Zig says: square of {d} is {d}\n", .{ num, square(num) });

    talk_from_c();
}
